plugins/github.js

```diff
@@ -1,9 +1,86 @@
 'use strict';
 const githubWebhooks = require('hapi-github-webhooks');
 const hoek = require('hoek');
+const async = require('async');
 const boom = require('boom');
 const Models = require('screwdriver-models');
 let Pipeline;
+let Job;
+let Build;
+
+/**
+ * Create a new job and start the build for an opened pull-request
+ * @method pullRequestOpened
+ * @param  {Object}       options
+ * @param  {String}       options.eventId    Unique ID for this GitHub event
+ * @param  {String}       options.pipelineId Identifier for the Pipeline
+ * @param  {String}       options.jobId      Identifier for the Job
+ * @param  {String}       options.name       Name of the new job (PR-1)
+ * @param  {Hapi.request} request Request from user
+ * @param  {Hapi.reply}   reply   Reply to user
+ */
+function pullRequestOpened(options, request, reply) {
+    const eventId = options.eventId;
+    const pipelineId = options.pipelineId;
+    const jobId = options.jobId;
+    const name = options.name;
+
+    async.waterfall([
+        // Create job
+        async.apply(Job.create.bind(Job), { pipelineId, name }),
+        // Log it
+        (job, next) => {
+            request.log(['webhook-github', eventId, jobId], `${job.name} created`);
+            next();
+        },
+        // Create build
+        async.apply(Build.create.bind(Build), { jobId }),
+        // Log it
+        (build, next) => {
+            request.log(['webhook-github', eventId, jobId, build.id], `${name} started `
+                + `${build.number}`);
+            next();
+        }
+    ], (waterfallError) => {
+        if (waterfallError) {
+            return reply(boom.wrap(waterfallError));
+        }
+
+        return reply().code(201);
+    });
+}
+
+/**
+ * Start any running builds and disable the job for an opened pull-request
+ * @method pullRequestClosed
+ * @param  {Object}       options
+ * @param  {String}       options.eventId    Unique ID for this GitHub event
+ * @param  {String}       options.pipelineId Identifier for the Pipeline
+ * @param  {String}       options.jobId      Identifier for the Job
+ * @param  {String}       options.name       Name of the job (PR-1)
+ * @param  {Hapi.request} request Request from user
+ * @param  {Hapi.reply}   reply   Reply to user
+ */
+function pullRequestClosed(options, request, reply) {
+    const eventId = options.eventId;
+    const jobId = options.jobId;
+    const name = options.name;
+
+    // @TODO stop running build
+    Job.update({
+        id: jobId,
+        data: {
+            state: 'DISABLED'
```

Comment:
I don't think we should delete the job, we want to keep the history around.  I'd be interested in a "MERGED" state.